CMakeLists.txt

```
@@ -21,6 +21,9 @@ else()
     set(MINGW FALSE)
 endif()
 
+set(TVOS (CMAKE_SYSTEM_NAME STREQUAL "tvOS"))
+set(WATCHOS (CMAKE_SYSTEM_NAME STREQUAL "watchOS"))
```

Does this actually work as intended, as in, the expressions evaluate to either TRUE or FALSE (or ON or OFF)? cmake's documentation doesn't mention that it does, as far as I can tell.

A perl one-liner run across the cmake-based projects on my local system doesn't show up anything similar-looking, and I don't remember having seen this construct before.

When in doubt, consider rewriting them as generator expressions:
```suggestion
set(TVOS $<STREQUAL:${CMAKE_SYSTEM_NAME},tvOS>)
set(WATCHOS $<STREQUAL:${CMAKE_SYSTEM_NAME},watchOS>)
```

---

You're right. I've debugged this and with the new code it's explicit and definitely correct.